plugins/action/action.h

```diff
@@ -25,9 +25,17 @@ class Action : public PluginBase
 {
 public:
     /**
-     * @brief Constructor (internal use only).
+     * @brief Constructor. Creates the plugin and associates it with a specified Device.
      *
-     * @param impl Private internal implementation.
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto action = std::make_shared<Action>(&device);
+     *     ```
+     *
+     * The plugin is owned by the device (and will be destroyed along with it).
```

Comment:
I would say: `Each plugin is specific to the device with which it got instantiated.`

I don't think the plugin is owned by the device because the plugin only gets destroyed when all shared pointers are deleted / out of scope.

Reply:
Thank you. That is a valid point - I have clearly not spent enough time with shared pointers.

Reply:
@julianoes I have updated the docs to remove the "owned/destroyed" stuff and simplify description for all plugins. I think this is enough.